packages/caliper-core/test/worker/tx-observers/internal-tx-observer.js

```diff
@@ -0,0 +1,296 @@
+            clock.tick(DEFAULT_UPDATE_INTERVAL);
+            expect(messengerMock.send).to.have.been.calledOnce;
+
+            clock.tick(DEFAULT_UPDATE_INTERVAL);
+            expect(messengerMock.send).to.have.been.calledTwice;
+
+            sinon.assert.notCalled(warnLogger);
+            sinon.assert.notCalled(errorLogger);
+        });
+
+        it('should not accumulate transactions finished before activation gracefully', async () => {
+            const observer = new InternalTxObserver(messengerMock, managerUuid, workerIndex);
+            observer.txFinished({ totalFinished: 1 });
+
+            await observer.activate(1, 'round1');
+
+            // Fast forward the clock
+            clock.tick(DEFAULT_UPDATE_INTERVAL);
+
+            // Check that the finished transactions are included in the update
+            const messageSent = messengerMock.send.getCall(0).args[0];
+            expect(messageSent.content.stats.txCounters.totalFinished).to.equal(0);
+            // Clean up
+            await observer.deactivate();
+        });
+
+        it('should handle concurrent transaction submissions correctly', async () => {
+            const observer = new InternalTxObserver(messengerMock, managerUuid, workerIndex);
+
+            await observer.activate(1, 'round1');
+
+            // Simulate concurrent transaction submissions
+            for (let i = 0; i < 100; i++) {
+                observer.txSubmitted(1);
+            }
+
+            // Fast forward the clock to trigger the periodic update
+            clock.tick(DEFAULT_UPDATE_INTERVAL);
+
+            // Check that the totalSubmitted is 100
+            const messageSent = messengerMock.send.getCall(0).args[0];
+            expect(messageSent.content.stats.txCounters.totalSubmitted).to.equal(100);
+
+            // Clean up
+            await observer.deactivate();
+        });
+
+        it('should send the correct update message with the correct transaction count during periodic updates', async () => {
+            const observer = new InternalTxObserver(messengerMock, managerUuid, workerIndex);
+            await observer.activate(1, 'round1');
+            observer.txSubmitted(12);
+
+            clock.tick(DEFAULT_UPDATE_INTERVAL);
+            sinon.assert.calledOnce(messengerMock.send);
+
+            const messageSent = messengerMock.send.getCall(0).args[0];
+            expect(messageSent).to.be.instanceOf(TxUpdateMessage);
+
+            expect(messageSent.sender).to.equal('mocked-messenger-uuid');
+            expect(messageSent.recipients).to.deep.equal([managerUuid]);
+            expect(messageSent.content.stats.txCounters.totalSubmitted).to.equal(12);
+            expect(messageSent.content.type).to.equal('txUpdate');
+        });
+
+        it('should handle very large updateInterval without sending updates prematurely', async () => {
+            const VERY_LARGE_INTERVAL = 24 * 60 * 60 * 1000; // 24 hours in milliseconds
```

Comment:
do you think it might be better to say 
```javascript
const HOUR_INTERVAL=60 * 60 * 1000;
sinon.stub(ConfigUtil, 'get').withArgs(ConfigUtil.keys.Observer.Internal.Interval).returns(24 * HOUR_INTERVAL)
```